ipset_test.go

```diff
@@ -624,3 +624,35 @@ func TestPointLess(t *testing.T) {
 	}
 
 }
+
+func TestIPSetEqual(t *testing.T) {
+	a := new(IPSet)
+	b := new(IPSet)
+
+	assertEqual := func(want bool) {
+		t.Helper()
+		if got := a.Equal(b); got != want {
+			t.Errorf("%v.Equal(%v) = %v want %v", a, b, got, want)
+		}
+	}
+
+	a.Add(MustParseIP("1.1.1.0"))
+	a.Add(MustParseIP("1.1.1.1"))
+	a.Add(MustParseIP("1.1.1.2"))
+	b.AddPrefix(MustParseIPPrefix("1.1.1.0/31"))
+	b.Add(MustParseIP("1.1.1.2"))
+	assertEqual(true)
+
+	a.RemoveSet(a)
```

Comment:
This effectively zeros `a`. But the underlying representation contains non-empty `in` and `out`. Fixing this is in part the topic of #109 and #110 and a decision about when to minimize the representation.